### PR TITLE
feat: create reusable action confirmation component

### LINKED ISSUE #7: [FEAT] Create a Reusable Action Confirmation Component
Develop a reusable confirmation component to enable users to confirm their actions before proceeding. This feature will help prevent users from making mistakes by providing a confirmation step. The component should be responsive and reflect the state of the action (destructive or confirmation). 

### Purpose
The purpose of this component is to enable users to confirm their actions before proceeding, thus preventing unintended actions and mistakes. By providing a confirmation step, the component ensures that users are certain about their decisions, enhancing the overall user experience and reducing the likelihood of errors.

### Acceptance Criteria

**Requirements**
The component should receive the following props

1. Heading - The main heading or title of the confirmation dialog.
2. Description - A description or message providing details about the action.
3. Close Action - A function to close the confirmation dialog without any action
4. Cancel Action - A function to cancel the action and close the dialog.
5. Confirmation Action - A function to confirm the action and proceed.
6. Variants: There should be different variants (Success, Destructive, etc) 

**Tasks**
- [ ] Develop the UI according to how you see it on the design 
- [ ] The component must be responsive on all screen sizes as seen on the design
- [ ]  Implement props to receive heading, description, close action, cancel action, confirmation action,
-  [ ] Create separate variants - destructive, confirmation, success and error 
- [ ] There must be proper management of states
- [ ] Implement state management to handle the open/close state of the confirmation dialog.
- [ ]  Implement visual indicators for destructive actions (e.g., red buttons, warning icons).
- [ ] Implement visual indicators for confirmation actions (e.g., green buttons, check icons).
- [ ] Write unit tests to cover different states, variants and props.

### Note
- Ensure code quality and maintainability.
- Write unit tests to cover different states and props.
- Provide clear documentation and usage examples**.

**Deadline: Friday 10am**** 

### 
Images and Figma Link 
https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=2202-12521&m=dev
Screenshot 2024-07-17 124321

### PR DESCRIPTION
**Pull request by Feng-09**
# Description
Create a reusable action confirmation modal to enable users to confirm their actions before proceeding. This helps users avoid mistakes.

**Closes #7 **

# Changes proposed
Create Action-confirmation folder in src/components/modals
Create Confirmation Modal component in src/components/modals/Action-confirmation/index.tsx
Create action-modal.test.tsx to run tests on Modal component

## What were you told to do?
Develop the UI according to the design
Make the component responsive on all screen sizes
Implement props to receive heading, description, close action, cancel action, confirmation action
Create variants - basic information, confirmation, destructive confirmation, success and error to render the respective modals
Implement visual indicators for confirmation actions
Write unit tests to cover different states, variants and props

[FEAT] Create a Reusable Action Confirmation Component

## What did you do?
I developed the UI according to the design
I made the component responsive on all screen sizes
I implemented props to receive heading, description, close action, cancel action, confirmation action and trigger message to display on the modal trigger
I created variants - basic information, confirmation, destructive confirmation, success and error to render the respective modals
I implemented visual indicators for confirmation actions
I wrote unit tests to cover different states, variants and props

I changed the dimensions and the positioning of the "X" close button in the dialog.tsx component to match the design. I believe everyone working with this component would also need this change.

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [X] My code follows the code style of this project.
- [X] This PR does not contain plagiarized content.
- [X] The title and description of the PR is clear and explains the approach.
- [X] I am making a pull request against the **dev branch** (left side).
- [X] My commit messages styles matches our requested structure.
- [X] My code additions will fail neither code linting checks nor unit test.
- [X] I am only making changes to files I was requested to.

# Screenshots/Videos
Screenshot 2024-07-20 221001
Screenshot 2024-07-19 100900

### CODE REVIEW

**Review by prudentbird — DISMISSED** [review]
Do not use svgs, use the lucide react icon library provided to you n

**Review by Feng-09 — COMMENTED** [review]
I removed the svgs from the public folder and used lucide react icons in my component

File: public/close-modal-icon.svg
Comment (prudentbird) [inline public/close-modal-icon.svg:1]: This svg is part of lucide react

**Review by prudentbird — CHANGES_REQUESTED** [review]

File: public/close-modal-icon.svg
Comment (Feng-09) [reply public/close-modal-icon.svg:1]: I removed it and imported it from lucide react

**Review by Feng-09 — COMMENTED** [review]
I am no longer using any svgs in the component. Only lucide components

**Review by prudentbird — CHANGES_REQUESTED** [review]
Use icons from lucide react

File: public/close-modal-icon.svg
Comment (prudentbird) [inline public/close-modal-icon.svg:1]: Delete this file

**Review by prudentbird — DISMISSED** [review]

**Review by prudentbird — APPROVED** [review]

**Pull request merged**